sunpy/instr/goes.py

```diff
@@ -583,16 +603,500 @@ def _goes_get_chianti_em(longflux, temp, satellite=8, abundances="coronal",
+        measure values, these are also generated and added to goeslc_new
+        using goes_chianti_tem() (See documentation for that function.)
+
+    Examples
+    --------
+    >>> from sunpy.lightcurve as lc
+    >>> goeslc = lc.GOESLightCurve.create(time1, time2)
+    >>> goeslc.data
+                               xrsa   xrsb
+    2014-01-01 00:00:00  9.1873e-08  4e-06
+    2014-01-01 00:00:02  9.1873e-08  4e-06
+    2014-01-01 00:00:04  9.1873e-08  4e-06
+    2014-01-01 00:00:06  9.2988e-08  4e-06
+    2014-01-01 00:00:08  9.1873e-08  4e-06
+    >>> goeslc_new = rad_loss_rate(goeslc)
+    >>> goeslc_new.data
+                        xrsa xrsb temperature            em  rad_loss_rate
+    2014-01-01 00:00:00  ...  ...    6.270239  6.440648e+48 5.44914366e+26
+    2014-01-01 00:00:02  ...  ...    6.270239  6.440648e+48 5.44914366e+26
+    2014-01-01 00:00:04  ...  ...    6.273917  6.422207e+48 5.44914366e+26
+    2014-01-01 00:00:06  ...  ...    6.304001  6.350370e+48 5.44914366e+26
+    2014-01-01 00:00:08  ...  ...    6.277604  6.403795e+48 5.44914366e+26
+
+    """
+    # Check that input argument is of correct type
+    if not isinstance(goeslc, lightcurve.GOESLightCurve):
+        raise TypeError("goeslc must be a GOESLightCurve object.")
+
+    # extract temperature and emission measure from GOESLightCurve
+    # object and change type to that required by calc_rad_loss().
+    # If GOESLightCurve object does not contain temperature and
+    # emission measure, calculate using temp_em()
+    if 'temperature' in goeslc.data and 'em' in goeslc.data:
+        goeslc_new = copy.deepcopy(goeslc)
+    else:
+        goeslc_new = temp_em(goeslc)
+    temp = Quantity(np.asarray(goeslc_new.data.temperature, dtype=np.float64), unit=units.MK)
+    em = Quantity(np.asarray(goeslc_new.data.em, dtype=np.float64), unit=units.cm**(-3))
+
+    # Find radiative loss rate with calc_rad_loss()
+    rad_loss_out = calc_rad_loss(temp, em, force_download=force_download,
+                                 download_dir=download_dir)
+
+    # Enter results into new version of GOES LightCurve Object
+    goeslc_new.data["rad_loss_rate"] = rad_loss_out["rad_loss_rate"]
+
+    return goeslc_new
+
+def calc_rad_loss(temp, em, obstime=None, force_download=False,
+                  download_dir=DATA_PATH):
+    """
+    Finds radiative loss rate of coronal plasma over all wavelengths.
+
+    This function calculates the radiative loss rate of solar coronal
+    soft X-ray-emitting plasma across all wavelengths given an isothermal
+    temperature and emission measure.  The units of the results are
+    erg/s.  This function is based on calc_rad_loss.pro in SSW IDL.
+    In addition, if obstime keyword is set, giving the times to which
+    the temperature and emission measure values correspond, the
+    radiated losses integrated over time are also calculated.
+
+    Parameters
+    ----------
+    temp : astropy Quantity
```

Comment:
references to `Quantity` should be like this `astropy.units.Quantity` which will make sphinx link to the astropy documentation for `Quantity`. Also it should read  'An `astropy.units.Quantity` instance.'  This goes for all the docstrings and documentation.

P.S. While I would not recommend it for this particular situation, you can precede a object reference like that with a tilde e.g. `~astropy.units.Quantity` which will generate the link but only print out the object name in this case 'Quantity'.